### PR TITLE
[Needs Testing][EE JIT] Creation of an EE Dynarec

### PR DESCRIPTION
**Pull request by Souzooka**
Just posting this branch for visibility purposes, this is the start to creating an EE dynamic recompiler. It's not ready to be merged at this point (or probably any time soon), but as of right now the recompiler detects branches, then builds a block which continuously falls back to the interpreter. It cannot handle branching or changes in the program counter itself for right now, but theoretically it should execute the very first instructions before a branch... over and over and over again.

### CODE REVIEW

**Comment by Souzooka** [comment]
Now the JIT has implemented the bare minimum of instructions needed to run PS2 programs (all branches, jumps, ERET, and SYSCALL). Some games get in game, but many still fail or have compatibility issues. Please comment on any issues you get with games that are different than the interpreter (additionally, please don't post about JitCache issues specifically. These can be circumvented by increasing the BLOCK_SIZE and POOL_SIZE variables in JitCache.hpp, but a new solution will be applied later).

Kingdom Hearts 2

**Comment by kenshen112** [comment]
I think this is perfectly merge-able, the compatibility's basically at the level of the Interpreter and the Gui switcher between the two covers the cases where either excels  -weirdbeard

File: src/core/ee/ee_jittrans.hpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
+#ifndef VU_JITTRANS_HPP
+#define VU_JITTRANS_HPP
```
Comment (phire) [inline src/core/ee/ee_jittrans.hpp]: wrong #define?

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,6 +18,14 @@ set(SOURCES
         src/core/ee/cop0.cpp
         src/core/ee/cop1.cpp
         src/core/ee/dmac.cpp
+        src/core/ee/ee_jit.cpp
+        src/core/ee/ee_jit64.cpp
+	src/core/ee/ee_jit64_mmi.cpp
```
Comment (Margen67) [inline CMakeLists.txt]: Is this supposed to be formatted differently from the rest of them?

**Pull request merged**